### ISSUE TITLE
Add Support for SBOM Diff

### ISSUE THREAD

**Issue by msymons** [CycloneDX/cyclonedx-cli#31]
It would be useful for the tool to be able to compare two SBOM and generate a diff report.

This would be handy for reviewing upgrades quickly, especially if the output can (optionally?) be in a "terse" format.

The diff report is something that could then be attached to a PR as evidence.  

Use case: I have maven projects that use dropwizard-BOM POM import in dependencyManagent.  Thus, a commit that tweaks nothing at all except the dropwizard version can have a big impact on dependencies and transitive dependencies...  all of which would be revealed in an SBOM  diff.

**Comment by coderpatros** [comment]
For initial implementation I'm thinking something like this...

```
cyclonedx-cli diff components --from-file software-v1.0.xml --to-file software-v1.1.xml

- SomeComponent@v2.0
+ SomeComponent@2.5
```

Other use cases might include checking for changes in dependency licenses.

Maybe something like this...

```
cyclonedx-cli diff licenses --from-file software-v1.0.xml --to-file software-v1.1.xml

+ GPL-3.0-or-later
```

And this feature, especially the license diff, might be well served by a fail on changes parameter.

Thinking something like this running as part of a "license check" for PRs...

```
cyclonedx-cli diff licenses --from-file previous-release.xml --to-file current-build.xml --fail-on-changes

+ GPL-3.0-or-later
```

**Comment by coderpatros** [comment]
Another thought, maybe this should focus more on new things.

i.e. `--new-supplier`, `--new-component`, `--new-component-version`, `--new-license` options?

**Comment by msymons** [comment]
Depending on how the output is formatted, a simple component diff could also be used to identify new components....right?

All of the options identified are useful.  I think that usage will depend on the situation.  For instance, when working on a development branch of a project, one would want to know all about license diffs.  When comparing a BOM from (say) an "Integration Environment" with what is running in "Production Environment", then I would very much want to know all about component diffs.  How out of date is production?

**Comment by coderpatros** [comment]
Initial support for component version diffs is available in v0.9.0
